src/transport_ofi.c

```diff
@@ -297,7 +297,15 @@ static inline int allocate_endpoints(struct fabric_info *info)
      * support shmem_wait that blocks on an update arriving from the network */
 #ifndef ENABLE_HARD_POLLING
     info->p_info->caps |= FI_REMOTE_WRITE | FI_REMOTE_READ;
+#else
+    info->p_info->caps &= ~FI_REMOTE_WRITE;
+    info->p_info->caps &= ~FI_REMOTE_READ;
 #endif
+
+    /* Remove unused capabilities */
+    info->p_info->caps &= ~FI_MSG;
+    info->p_info->caps &= ~FI_TAGGED;
```

Comment:
but then whats the point of turning them off? either its a provider bug if having them on the caps changes functionality from what you asked the provider for or there is nothing to gain from turning them off; that's just my two cents